apps/avifgainmaputil/tonemap_command.cc

```diff
@@ -90,66 +79,121 @@ avifResult TonemapCommand::Run() {
-    }
-    clli_set = (clli_box.maxCLL != 0) || (clli_box.maxPALL != 0);
+
+  // We are either tone mapping to the base image (i.e. leaving it as is),
+  // or tone mapping to the alternate image (i.e. fully applying the gain map),
+  // or tone mapping in between (partially applying the gain map).
+  const bool tone_mapping_to_base =
+      (headroom <= metadata.baseHdrHeadroom &&
+       metadata.baseHdrHeadroom <= metadata.alternateHdrHeadroom) ||
+      (headroom >= metadata.baseHdrHeadroom &&
+       metadata.baseHdrHeadroom >= metadata.alternateHdrHeadroom);
+  const bool tone_mapping_to_alternate =
+      (headroom <= metadata.alternateHdrHeadroom &&
+       metadata.alternateHdrHeadroom <= metadata.baseHdrHeadroom) ||
+      (headroom >= metadata.alternateHdrHeadroom &&
+       metadata.alternateHdrHeadroom >= metadata.baseHdrHeadroom);
+  const bool base_is_hdr = (metadata.baseHdrHeadroom != 0.0f);
+
+  // Determine output CICP.
+  CicpValues cicp;
+  if (arg_output_cicp_.provenance() == argparse::Provenance::SPECIFIED) {
+    cicp = arg_output_cicp_;  // User provided values.
+  } else if (tone_mapping_to_base || (tone_mapping_to_hdr && base_is_hdr)) {
+    cicp = {image->colorPrimaries, image->transferCharacteristics,
+            image->matrixCoefficients};
+  } else {
+    cicp = {image->gainMap->altColorPrimaries,
+            image->gainMap->altTransferCharacteristics,
+            image->gainMap->altMatrixCoefficients};
+  }
+  if (cicp.color_primaries == AVIF_COLOR_PRIMARIES_UNSPECIFIED) {
+    // TODO(maryla): for now avifImageApplyGainMap always uses the primaries of
+    // the base image, but it should take into account the metadata's
+    // useBaseColorSpace property.
+    cicp.color_primaries = image->colorPrimaries;
+  }
+  if (cicp.transfer_characteristics ==
+      AVIF_TRANSFER_CHARACTERISTICS_UNSPECIFIED) {
+    cicp.transfer_characteristics =
+        tone_mapping_to_hdr ? AVIF_TRANSFER_CHARACTERISTICS_SMPTE2084
+                            : AVIF_TRANSFER_CHARACTERISTICS_SRGB;
   }
 
+  // Determine output depth.
   int depth = arg_image_read_.depth;
   if (depth == 0) {
-    depth = tone_mapping_to_sdr ? 8 : decoder->image->depth;
+    if (tone_mapping_to_base) {
+      depth = image->depth;
+    } else if (tone_mapping_to_alternate) {
+      depth = image->gainMap->altDepth;
+    }
+    if (depth == 0) {
+      // Default to the max depth between the base image, the gain map and
+      // the specified 'altDepth'.
+      depth = std::max(std::max(image->depth, image->gainMap->image->depth),
+                       image->gainMap->altDepth);
+    }
   }
+
+  // Determine output pixel format.
   avifPixelFormat pixel_format =
       (avifPixelFormat)arg_image_read_.pixel_format.value();
+  const avifPixelFormat alt_yuv_format =
+      (image->gainMap->altPlaneCount == 1)
+          ? AVIF_PIXEL_FORMAT_YUV400
+          : std::min(image->yuvFormat, image->gainMap->image->yuvFormat);
```

Comment:
`std::min()` on an enum is weird but it works.  
Maybe add a comment `// Favor least subsampled format` to avoid having to look at the `avifPixelFormat` definition? Or is it better to leave it like this to avoid disconnection between `avifPixelFormat` order and the comment?